core/chains/chain_set.go

```diff
@@ -22,14 +22,16 @@ var (
 	ErrNotFound     = errors.New("not found")
 )
 
-// Chains is a generic interface for chain configuration.
-type Chains interface {
+// ChainStatuser is a generic interface for chain configuration.
+type ChainStatuser interface {
+	// must return [ErrNotFound] if the id is not found
 	ChainStatus(ctx context.Context, id string) (types.ChainStatus, error)
 	ChainStatuses(ctx context.Context, offset, limit int) ([]types.ChainStatus, int, error)
```

Comment:
Out of scope for this PR, but documentation here would be helpful to me. How are `offset` and `limit` related to Chain Statuses? When concrete types implicitly satisfy this interface, what are some basic rules for how `offset` and `limit` can be used?